python/lsst/sims/catUtils/utils/SNIaLightCurveGenerator.py

```diff
@@ -0,0 +1,164 @@
+                    t_max = local_t_max
+
+        snobj = SNObject()
+
+        cat = cat_dict[cat_dict.keys()[0]]  # does not need to be associated with a bandpass
+
+        for chunk in query_result:
+            t_start_chunk = time.time()
+            for sn in cat.iter_catalog(query_cache=[chunk]):
+                sn_rng = self.sn_universe.getSN_rng(sn[1])
+                sn_t0 = self.sn_universe.drawFromT0Dist(sn_rng)
+                if sn[5] <= self.z_cutoff and np.isfinite(sn_t0) and \
+                    sn_t0 < t_max + cat.maxTimeSNVisible and \
+                    sn_t0 > t_min - cat.maxTimeSNVisible:
+
+                    sn_c = self.sn_universe.drawFromcDist(sn_rng)
+                    sn_x1 = self.sn_universe.drawFromx1Dist(sn_rng)
+                    sn_x0 = self.sn_universe.drawFromX0Dist(sn_rng, sn_x1, sn_c, sn[4])
+
+                    snobj.set(t0=sn_t0, c=sn_c, x1=sn_x1, x0=sn_x0, z=sn[5])
+
+                    for bp_name in cat_dict:
+                        t_list = t_dict[bp_name]
+                        m5_list = m5_dict[bp_name]
+                        gamma_list = gamma_dict[bp_name]
+                        bandpass = self.lsstBandpassDict[bp_name]
+                        if len(t_list) == 0:
+                            continue
+
+                        if snobj.maxtime() >= t_list[0] and snobj.mintime() <= t_list[-1]:
+                            active_dexes = np.where(np.logical_and(t_list >= snobj.mintime(),
+                                                                   t_list <= snobj.maxtime()))
+
+                            t_active = t_list[active_dexes]
+                            m5_active = m5_list[active_dexes]
+                            gamma_active = gamma_list[active_dexes]
+
+                            if len(t_active) > 0:
+                                wave_ang = bandpass.wavelen*10.0
+                                mask = np.logical_and(wave_ang > snobj.minwave(),
+                                                      wave_ang < snobj.maxwave())
+
+                                wave_ang = wave_ang[mask]
+                                snobj.set(mwebv=sn[6])
+                                sn_ff_buffer = snobj.flux(time=t_active, wave=wave_ang)*10.0
+                                flambda_grid = np.zeros((len(t_active), len(bandpass.wavelen)))
+                                for ff, ff_sn in zip(flambda_grid, sn_ff_buffer):
+                                    ff[mask] = np.where(ff_sn > 0.0, ff_sn, 0.0)
+
+                                ss = Sed()
+                                fnu_grid = flambda_grid*bandpass.wavelen* \
+                                           bandpass.wavelen*ss._physParams.nm2m* \
+                                           ss._physParams.ergsetc2jansky/ss._physParams.lightspeed
+
+                                flux_list = \
+                                (fnu_grid*bandpass.phi).sum(axis=1)*(bandpass.wavelen[1]-bandpass.wavelen[0])
```

Comment:
It might be faster to use `np.dot`, I often find that to be faster than `np.sum()`.  Also the $d\lambda$  term is probably `bandpass.step`. You should be able to do the calculation from flambda_grid to fluxes/magnitudes directly without going to fnu.